components/payments/cmd/connectors/internal/connectors/modulr/task_fetch_accounts.go

```diff
@@ -19,13 +19,60 @@ import (
 	"go.opentelemetry.io/otel/attribute"
 )
 
+type fetchAccountsState struct {
+	LastCreatedTime time.Time `json:"last_created_date"`
+}
+
+func (s *fetchAccountsState) UpdateLatest(latest *client.Account) error {
+	createdTime, err := time.Parse("2006-01-02T15:04:05.999-0700", latest.CreatedDate)
+	if err != nil {
+		return err
+	}
+	if createdTime.After(s.LastCreatedTime) {
+		s.LastCreatedTime = createdTime
+	}
+	return nil
+}
+
+func (s *fetchAccountsState) IsNew(account *client.Account) (bool, error) {
+	createdTime, err := time.Parse("2006-01-02T15:04:05.999-0700", account.CreatedDate)
+	if err != nil {
+		return false, err
+	}
+	return createdTime.After(s.LastCreatedTime), nil
+}
+
+func (s *fetchAccountsState) FilterNew(accounts []*client.Account) ([]*client.Account, error) {
```

Comment:
This function is not working when there is no new element in the slice. The firstNewIdx will never be updated and will still be 0, so you will process again all accounts of the slice

Reply:
fixed. This index is now initialized with the slice's length.